### ISSUE TITLE
Attempted to read or write protected memory - Very Serious Error :-(

### ISSUE THREAD

**Issue by TaviTruman** [Microsoft/GraphEngine#382]
System.AccessViolationException
  HResult=0x80004003
  Message=Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
  
I have about 100 or so items in the local memory cloud. This application is a WPF app using TGE for a local (Embedded) Knowledge graph. 

This is the second time I have encountered this problem in the past week or so. I have been using the TGE for ten years now and I have never had this type of low-level memory management error. 

Here is the offending C# code calling in the TGE API:

![image](https://github.com/microsoft/GraphEngine/assets/5692812/81461079-2ea3-470b-b487-75fcd872b382)

Here is the corresponding TSL Schema Code:

![image](https://github.com/microsoft/GraphEngine/assets/5692812/1f7f62d1-55fc-44bb-95ae-f7645decc4b4)

There error as far as I tell occurs on line 3343:

![image](https://github.com/microsoft/GraphEngine/assets/5692812/888bd854-6192-4140-83ee-035041be35e1)

**Comment by TaviTruman** [comment]
I have uploaded the C# part of the app that encounters the error and the part of the TSL Schema
[TGE-Write Protected Memory Error.zip](https://github.com/microsoft/GraphEngine/files/13401826/TGE-Write.Protected.Memory.Error.zip)

**Comment by TaviTruman** [comment]
Okay, I have more research and the problem stems from a corrupted TGE Image source file on disk. I took a look at the TGE log files and I did see something strange; to remedy the problem I deleted the TGE image sources and reran my test and this error did not reoccur. I need to do some more systems engineering design work on the hardware (storage) side of things as well as make some changes to the "storage" configuration for GE.
